### PR TITLE
OPENTOK-49599: Bluetooth issues on macOS SDK

### PR DESCRIPTION
**Pull request by jvalli**
#### What is this PR doing?
Add support bluetooth audio device support on MacOS SDK custom audio sample app

#### How should this be manually tested?
- Create a session
- Start publishing and subscribing
- Add the bluetooth audio device

#### What are the relevant tickets?

[OPENTOK-49599](https://jira.vonage.com/browse/OPENTOK-49599)

### CODE REVIEW

File: Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m [hunk 1]
```diff
@@ -711,4 +736,202 @@ - (BOOL)setPlayOutRenderCallback:(AudioUnit)unit
...
+{
+    AudioObjectPropertyAddress address = {
+        kAudioDevicePropertyStreamConfiguration,
+        scope,
+        kAudioObjectPropertyElementMain,
+    };
+
+    AudioBufferList streamConfiguration;
+    UInt32 propSize = sizeof(streamConfiguration);
+    OSStatus status = AudioObjectGetPropertyData(deviceID,
```
Comment (juliobecerragomez) [inline Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m:849]: Should we call `AudioObjectGetPropertyDataSize` to know the proper size before allocating streamConfiguration? Or this is enough?

File: Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m [hunk 2]
```diff
@@ -711,4 +736,202 @@ - (BOOL)setPlayOutRenderCallback:(AudioUnit)unit
...
+    if (status == noErr) {
+        return [NSString stringWithFormat:@"No error (%d)", status];
+    }
+
+    return [NSString stringWithFormat:@"Error \"%s\" (%d)",
+            codeToString(status),
+            status];
+}
+
+static char *codeToString(UInt32 code)
```
Comment (juliobecerragomez) [inline Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m:812]: Out of curiosity, why is this needed? It seems to be only used to copy bytes from the uint32 to the string. Then it is always used together with the uint32.

**Review by juliobecerragomez — COMMENTED** [review]
Looks good! just a couple of questions

File: Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m (hunk 1 above)
Comment (jvalli) [reply Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m:849]: On my testing this was enough.

File: Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m (hunk 2 above)
Comment (jvalli) [reply Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m:812]: It's just for logging purposes.

**Review by juliobecerragomez — APPROVED** [review]
LGTM!

File: Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m [hunk 3]
```diff
@@ -87,6 +87,31 @@ - (instancetype)init
...
+                        break;
+                    default:
+                        break;
+                }
+                index++;
+            }
+        };
+        
+        OSStatus status = AudioObjectAddPropertyListenerBlock(kAudioObjectSystemObject, &devicePropertyAddress, nil, audioObjectPropertyListenerBlock);
+        NSError *error = [NSError errorWithDomain:NSOSStatusErrorDomain code:status userInfo:nil];
```
Comment (goncalocostamendes) [inline Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m:113]: I don't understand why we are raising this error always. Probably it is my objc limitations the reason why I can't see

Comment (jvalli) [reply Custom-Audio-Driver/Custom-Audio-Driver/OTDefaultAudioDevice-Mac.m]: Updated!
